packages/clay-modal/src/Footer.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React, {FunctionComponent, HTMLAttributes} from 'react';
+import warning from 'warning';
+
+export interface FooterContainerProps extends HTMLAttributes<HTMLDivElement> {
+	position?: number;
+}
+
+const ButtonWrap: FunctionComponent<HTMLAttributes<HTMLDivElement>> = ({
+	children,
+}) => <div className="btn-group-item">{children}</div>;
+
+const FooterContainer: FunctionComponent<FooterContainerProps> = ({
+	children,
+	position,
+}) => (
+	<div
+		className={classNames({
+			'modal-item-first': position === 0,
+			'modal-item-last': position === 1,
+		})}
+	>
+		<div className="btn-group">
+			{React.Children.map(children, (child, i) =>
+				React.cloneElement(<ButtonWrap>{child}</ButtonWrap>, {key: i})
+			)}
+		</div>
+	</div>
+);
+
+interface FooterProps {
+	children: React.ReactElement<FunctionComponent<FooterContainerProps>>;
+}
+
+/**
+ * Handles the childrens of the footer to format the markup and add
+ * the necessary classNames.
+ */
+const Footer: FunctionComponent<FooterProps> = ({children}) => {
+	let childrens = children;
```

Comment:
I think it would not be necessary to unwrap all Fragments, the first case is necessary just not to break the markup. Also not sure if anyone would use more than one fragment here.